internal/usecases/reports/get_assets_report.go

```diff
@@ -0,0 +1,153 @@
+package reports
+
+import (
+	"context"
+	"github.com/rsksmart/liquidity-provider-server/internal/entities"
+	"github.com/rsksmart/liquidity-provider-server/internal/entities/blockchain"
+	"github.com/rsksmart/liquidity-provider-server/internal/entities/liquidity_provider"
+	"github.com/rsksmart/liquidity-provider-server/internal/entities/quote"
+	"github.com/rsksmart/liquidity-provider-server/pkg"
+)
+
+type GetAssetsReportUseCase struct {
+	btcWallet        blockchain.BitcoinWallet
+	rsk              blockchain.Rpc
+	lp               liquidity_provider.LiquidityProvider
+	peginProvider    liquidity_provider.PeginLiquidityProvider
+	pegoutProvider   liquidity_provider.PegoutLiquidityProvider
+	peginRepository  quote.PeginQuoteRepository
+	pegoutRepository quote.PegoutQuoteRepository
+}
+
+func NewGetAssetsReportUseCase(
+	wallet blockchain.BitcoinWallet,
+	rsk blockchain.Rpc,
+	lp liquidity_provider.LiquidityProvider,
+	peginProvider liquidity_provider.PeginLiquidityProvider,
+	pegoutProvider liquidity_provider.PegoutLiquidityProvider,
+	peginRepository quote.PeginQuoteRepository,
+	pegoutRepository quote.PegoutQuoteRepository,
+) *GetAssetsReportUseCase {
+	return &GetAssetsReportUseCase{
+		btcWallet:        wallet,
+		rsk:              rsk,
+		lp:               lp,
+		peginProvider:    peginProvider,
+		pegoutProvider:   pegoutProvider,
+		peginRepository:  peginRepository,
+		pegoutRepository: pegoutRepository,
+	}
+}
+
+func (useCase *GetAssetsReportUseCase) Run(ctx context.Context) (pkg.GetAssetsReportResponse, error) {
```

Comment:
don't return a DTO in the use case, return a domain object and then convert it to DTO in the handler